src/hotspot/share/gc/shenandoah/shenandoahHeapRegionCounters.cpp

```diff
@@ -75,17 +74,33 @@ ShenandoahHeapRegionCounters::ShenandoahHeapRegionCounters() :
                                                                PerfData::U_None, CHECK);
     }
 
-    if (ShenandoahLogRegionSampling) {
-      _log_file = new ShenandoahLogFileOutput(ShenandoahRegionSamplingFile, _timestamp->get_value());
-      _log_file->set_option(ShenandoahLogFileCount, ShenandoahLogFileSize);
-      _log_file->initialize(tty);
-    }
   }
 }
 
 ShenandoahHeapRegionCounters::~ShenandoahHeapRegionCounters() {
   if (_name_space != NULL) FREE_C_HEAP_ARRAY(char, _name_space);
-  if (_log_file != NULL) FREE_C_HEAP_OBJ(_log_file);
+}
+
+void ShenandoahHeapRegionCounters::write_snapshot(PerfLongVariable** regions,
+                                             PerfLongVariable* ts,
+                                             PerfLongVariable* status,
+                                             size_t num_regions,
+                                             size_t region_size, size_t protocol_version) {
+    LogTarget(Debug, gc, region) lt;
+    if (lt.is_enabled()) {
+        ResourceMark rm;
+        LogStream ls(lt);
+
+        ls.print_cr("%li %li %lu %lu %lu",
+                    ts->get_value(), status->get_value(), num_regions, region_size, protocol_version);
+        if (num_regions > 0) {
+            ls.print("%li", regions[0]->get_value());
+        }
+        for (uint i = 1; i < num_regions; ++i) {
+            ls.print(" %li", regions[i]->get_value());
+        }
+        ls.cr();
+    }
```

Comment:
I think the indenting style is 2 spaces.

Reply:
Thank you @shipilev . We have a PR open on the visualizer itself with changes that compliment these: https://github.com/openjdk/shenandoah-visualizer/pull/1. We would really appreciate it if you could find time to also review that PR.